### PR TITLE
Handle subprocess output.

### PR DESCRIPTION
**Pull request by jmcarp**
Subprocess returns a bytestring, so comparisons with strings in python
3.x throw unexpected errors. This patch decodes bytestrings to strings
and updates tests to return realistic output.

Here's some sample output when the target repo isn't found:

```
(cookiecutter)jmcarp@Joshuas-MBP-3|~/code/cookiecutter on master!
± cookiecutter https://github.com/not/exists
Traceback (most recent call last):
  File "/Users/jmcarp/miniconda/envs/cookiecutter/lib/python3.6/site-packages/cookiecutter/vcs.py", line 123, in clone
    stderr=subprocess.STDOUT,
  File "/Users/jmcarp/miniconda/envs/cookiecutter/lib/python3.6/subprocess.py", line 336, in check_output
    **kwargs).stdout
  File "/Users/jmcarp/miniconda/envs/cookiecutter/lib/python3.6/subprocess.py", line 418, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['git', 'clone', 'https://github.com/not/exists']' returned non-zero exit status 128.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/jmcarp/miniconda/envs/cookiecutter/bin/cookiecutter", line 11, in <module>
    load_entry_point('cookiecutter==1.5.1', 'console_scripts', 'cookiecutter')()
  File "/Users/jmcarp/miniconda/envs/cookiecutter/lib/python3.6/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/Users/jmcarp/miniconda/envs/cookiecutter/lib/python3.6/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/Users/jmcarp/miniconda/envs/cookiecutter/lib/python3.6/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/jmcarp/miniconda/envs/cookiecutter/lib/python3.6/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/Users/jmcarp/miniconda/envs/cookiecutter/lib/python3.6/site-packages/cookiecutter/cli.py", line 123, in main
    default_config=default_config,
  File "/Users/jmcarp/miniconda/envs/cookiecutter/lib/python3.6/site-packages/cookiecutter/main.py", line 63, in cookiecutter
    no_input=no_input,
  File "/Users/jmcarp/miniconda/envs/cookiecutter/lib/python3.6/site-packages/cookiecutter/repository.py", line 84, in determine_repo_dir
    no_input=no_input,
  File "/Users/jmcarp/miniconda/envs/cookiecutter/lib/python3.6/site-packages/cookiecutter/vcs.py", line 132, in clone
    if 'not found' in clone_error.output.lower():
TypeError: a bytes-like object is required, not 'str'
```

### CODE REVIEW

**Comment by denis-osipov** [comment]
This patch works, at least for my pyhon3 envs. Why it's not merged?

**Comment by hackebrot** [comment]
Hi @jmcarp! Thank you for this PR 🙇

Apologies for the late response.

LGTM 👍 Anyone else any feedback @michaeljoseph @audreyr @pydanny?

@denis-osipov Thank you for testing this PR on Python3! We haven't merged this just yet, because none of the core contributors are paid to work on this open source project. While we are grateful for all contributions, it sometimes just takes a bit more time than usual to review and merge changes, especially if other things like writing a book or moving countries are happening in our personal lives.

If you would like support this project and help us increase the speed in which we integrate new changes, please consider commenting on #920 if you or your company is happy to sponsor Cookiecutter.

Thanks! 🍪

**Comment by denis-osipov** [comment]
@hackebrot Thank you for your response. Sorry for my inaccurate wording. I didn't want neither to hurry you nor to complain. I just wanted to know if it's all right with this patch. I've been learning how to program for couple of weeks and I don't understand well how all this stuff works.

Anyway if the bug will be fixed it will be great!

**Comment by hackebrot** [comment]
🔔 @michaeljoseph

File: cookiecutter/vcs.py [hunk 1]
```diff
@@ -129,12 +129,13 @@ def clone(repo_url, checkout=None, clone_to_dir='.', no_input=False):
                 stderr=subprocess.STDOUT,
             )
     except subprocess.CalledProcessError as clone_error:
-        if 'not found' in clone_error.output.lower():
+        output = clone_error.output.decode()
```
Comment (michaeljoseph) [inline cookiecutter/vcs.py]: I think we're going to need to handle the py2 case here as well?

File: tests/vcs/test_clone.py [hunk 2]
```diff
@@ -139,8 +139,8 @@ def test_clone_should_invoke_vcs_command(
 
 
 @pytest.mark.parametrize('error_message', [
-    "fatal: repository 'https://github.com/hackebro/cookiedozer' not found",
-    'hg: abort: HTTP Error 404: Not Found',
+    b"fatal: repository 'https://github.com/hackebro/cookiedozer' not found",
+    b'hg: abort: HTTP Error 404: Not Found',
```
Comment (michaeljoseph) [inline tests/vcs/test_clone.py]: `subprocess.CalledProcessError.output` won't be a bytestring in py2...

File: cookiecutter/vcs.py (hunk 1 above)
Comment (michaeljoseph) [inline cookiecutter/vcs.py]: Could we be explicit about the encoding here please?

**Comment by jmcarp** [comment]
@michaeljoseph: updated.

**Comment by theodesp** [comment]
@jmcarp  Can you review all the changes requested for this ticket as some of the checks are missing.
https://github.com/audreyr/cookiecutter/pull/905/files/e693538a8278b289729de831181286c062266cbe#diff-8327dc289ba1c55ec5be92a9f27829ff

**Comment by jmcarp** [comment]
@theodesp: amended.

**Comment by theodesp** [comment]
@jmcarp Can you PM the team in Gitter to review your changes?
https://gitter.im/audreyr/cookiecutter

**Comment by michaeljoseph** [comment]
Thanks for the contribution @jmcarp 🍰

**Pull request merged**